index.js

```diff
@@ -12,28 +14,31 @@ function create(input, opts) {
 			return '---';
 		}
 
-		line = objectAssign(line, opts);
+		line = Object.assign(line, opts);
 
-		var text = String(line.text);
+		const text = String(line.text);
 		delete line.text;
 
 		if (!text) {
 			throw new Error('text required');
 		}
 
-		return text.split('\n').map(function (textLine) {
-			return textLine + '|' + Object.keys(line).map(function (x) {
-				if (x === 'href') {
-					x = encodeURI(x);
-				}
+		if (typeof line.submenus === 'object' && line.submenus.length > 0) {
+			submenuText = `\n${create(line.submenus, opts, menuLevel + 1)}`;
+			delete line.submenus;
+		}
+
+		const prefix = '--'.repeat(menuLevel);
 
-				return x + '="' + line[x] + '"';
-			}).join(' ');
-		}).join('\n');
+		return text.split('\n').map((textLine) => {
+			return `${prefix}${textLine}|${Object.keys(line).map((x) => {
```

Comment:
Don't put so much logic inside the template string. Do it outside and use a variable of the result inside.